js/ui/windowManager.js

```diff
@@ -1204,6 +1204,21 @@ function getAnimatableWindowActors() {
     ].indexOf(w.meta_window.get_window_type()) !== -1));
 }
 
+function getTaskbarIconGeometryForWindow(window) {
+    let actor = Shell.WindowTracker.get_default().get_window_app(window).taskbarIconActor;
+
+    if (!actor)
+        return null;
+
+    let [x, y] = actor.get_transformed_position();
+    return {
+        x1: x,
+        x2: x + actor.width,
+        y1: y,
+        y2: y + actor.height
```

Comment:
Why do you define `x1, x2, y1, y2` instead of `x1, y1, x2, y2` (which makes more sense if we're thinking about two different points)?

Reply:
Indeed. Fixed.